extra/modules/fiftyone-devicedetection/src/main/java/org/prebid/server/hooks/modules/fiftyone/devicedetection/v1/hooks/FiftyOneDeviceDetectionRawAuctionRequestHook.java

```diff
@@ -0,0 +1,282 @@
+import org.prebid.server.hooks.modules.fiftyone.devicedetection.model.config.AccountFilter;
+import org.prebid.server.hooks.modules.fiftyone.devicedetection.v1.adapters.DeviceMirror;
+import org.prebid.server.hooks.modules.fiftyone.devicedetection.v1.model.ModuleContext;
+import org.prebid.server.hooks.modules.fiftyone.devicedetection.v1.model.InvocationResultImpl;
+import org.prebid.server.hooks.v1.InvocationAction;
+import org.prebid.server.hooks.v1.InvocationResult;
+import org.prebid.server.hooks.v1.InvocationStatus;
+import org.prebid.server.hooks.v1.auction.AuctionInvocationContext;
+import org.prebid.server.hooks.v1.auction.AuctionRequestPayload;
+import org.prebid.server.hooks.v1.auction.RawAuctionRequestHook;
+import io.vertx.core.Future;
+import org.prebid.server.settings.model.Account;
+import org.prebid.server.util.ObjectUtil;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.BiConsumer;
+import java.util.function.BiFunction;
+import java.util.function.Consumer;
+import java.util.function.Predicate;
+
+public class FiftyOneDeviceDetectionRawAuctionRequestHook implements RawAuctionRequestHook {
+
+    private static final String CODE = "fiftyone-devicedetection-raw-auction-request-hook";
+
+    private final AccountFilter accountFilter;
+    private final DevicePatchPlanner devicePatchPlanner;
+    private final DeviceDetector deviceDetector;
+    private final DeviceInfoPatcher<Device> deviceInfoPatcher;
+
+    public Predicate<AuctionInvocationContext> accountControl = this::isAccountAllowed;
+    public BiConsumer<CollectedEvidenceBuilder, BidRequest> bidRequestEvidenceCollector = this::collectEvidence;
+    public BiConsumer<UserAgent, Map<String, String>> userAgentEvidenceConverter = this::appendSecureHeaders;
+    public BiFunction<ModuleContext, Consumer<CollectedEvidenceBuilder>, ModuleContext> moduleContextPatcher
+            = FiftyOneDeviceDetectionRawAuctionRequestHook::addEvidenceToContext;
+    public BiFunction<BidRequest, CollectedEvidence, BidRequest> bidRequestPatcher = this::enrichDevice;
+
+    public FiftyOneDeviceDetectionRawAuctionRequestHook(
+            AccountFilter accountFilter,
+            DevicePatchPlanner devicePatchPlanner,
+            DeviceDetector deviceDetector,
+            DeviceInfoPatcher<Device> deviceInfoPatcher)
+    {
+        this.accountFilter = accountFilter;
+        this.devicePatchPlanner = devicePatchPlanner;
+        this.deviceDetector = deviceDetector;
+        this.deviceInfoPatcher = deviceInfoPatcher;
+    }
+
+    @Override
+    public String code() {
+        return CODE;
+    }
+
+    @Override
+    public Future<InvocationResult<AuctionRequestPayload>> call(
+            AuctionRequestPayload payload,
+            AuctionInvocationContext invocationContext)
+    {
+        if (!accountControl.test(invocationContext)) {
```

Comment:
please just call the methods instead of wrapping them with the functional interfaces